# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1301630257 25200
#      Thu Mar 31 20:57:37 2011 -0700
# Node ID 0ade5992e3744e08c4f133ac30d0f2f94d1ff12a
# Parent  d6ad4ed57dda75604b46efd868ba39516d385b96
Add documentation for '-forge' option (bug #32464).

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2011-03-31  Rik  <octave@nomad.inbox5.com>
+
+	* pkg/module.mk, pkg/pkg.m, pkg/private/get_forge_pkg.m: Add
+	documentation for '-forge' option (bug #32464).  Make get_forge_pkg
+	a private function.
+
 2011-03-31  Marco Caliari  <marco.caliari@univr.it>
 
 	* sparse/spdiags.m: Treat empty vector (1x0 or 0x1) the same as diag().
 
 2011-03-31  Rik  <octave@nomad.inbox5.com>
 
 	* statistics/base/cor.m: Increase tolerance of %!tests using random
 	data to guarantee passage.
diff --git a/scripts/pkg/module.mk b/scripts/pkg/module.mk
--- a/scripts/pkg/module.mk
+++ b/scripts/pkg/module.mk
@@ -1,11 +1,14 @@
 FCN_FILE_DIRS += pkg
 
+pkg_PRIVATE_FCN_FILES = \
+  pkg/private/get_forge_pkg.m
+
 pkg_FCN_FILES = \
   pkg/pkg.m \
-  pkg/get_forge_pkg.m
+  $(pkg_PRIVATE_FCN_FILES)
 
 FCN_FILES += $(pkg_FCN_FILES)
 
 PKG_ADD_FILES += pkg/PKG_ADD
 
 DIRSTAMP_FILES += pkg/$(octave_dirstamp)
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -53,16 +53,20 @@
 ## starting Octave, even if the package requests that it isn't.
 ##
 ## @item -local
 ## A local installation is forced, even if the user has system privileges.
 ##
 ## @item -global
 ## A global installation is forced, even if the user doesn't normally have
 ## system privileges
+## 
+## @item -forge
+## Install a package directly from the Octave-Forge repository.  This
+## requires an internet connection and the cURL library.
 ##
 ## @item -verbose
 ## The package manager will print the output of all of the commands that are
 ## performed.
 ## @end table
 ##
 ## @item uninstall
 ## Uninstall named packages.  For example,
@@ -113,16 +117,19 @@
 ## @example
 ## [@var{user_packages}, @var{system_packages}] = pkg list
 ## @end example
 ##
 ## @noindent
 ## splits the list of installed packages into those who are installed by
 ## the current user, and those installed by the system administrator.
 ##
+## The option '-forge' lists packages available at the Octave-Forge repository.
+## This requires an internet connection and the cURL library.
+##
 ## @item describe
 ## Show a short description of the named installed packages, with the option
 ## '-verbose' also list functions provided by the package, e.g.:
 ##
 ## @example
 ##  pkg describe -verbose all
 ## @end example
 ##
diff --git a/scripts/pkg/get_forge_pkg.m b/scripts/pkg/private/get_forge_pkg.m
rename from scripts/pkg/get_forge_pkg.m
rename to scripts/pkg/private/get_forge_pkg.m
